sktime/classification/deep_learning/cnn.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""Time Convolutional Neural Network (CNN) for classification."""
+
+__author__ = ["James-Large", "TonyBagnall"]
+__all__ = ["CNNClassifier"]
+from sklearn.utils import check_random_state
+
+from sktime.classification.deep_learning.base import BaseDeepClassifier
+from sktime.networks.cnn import CNNNetwork
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("tensorflow", severity="warning")
+
+
+class CNNClassifier(BaseDeepClassifier, CNNNetwork):
+    """Time Convolutional Neural Network (CNN), as described in [1].
+
+    Parameters
+    ----------
+    should inherited fields be listed here?
+    n_epochs       : int, default = 2000
+        the number of epochs to train the model
+    batch_size      : int, default = 16
+        the number of samples per gradient update.
+    kernel_size     : int, default = 7
+        the length of the 1D convolution window
+    avg_pool_size   : int, default = 3
+        size of the average pooling windows
+    n_conv_layers   : int, default = 2
+        the number of convolutional plus average pooling layers
+    filter_sizes    : array of shape (n_conv_layers) default = [6, 12]
+    random_state    : int or None, default=None
+        Seed for random number generation.
+    verbose         : boolean, default = False
+        whether to output extra information
+    loss            : string, default="mean_squared_error"
+        fit parameter for the keras model
+    optimizer       : keras.optimizer, default=keras.optimizers.Adam(),
+    metrics         : list of strings, default=["accuracy"],
+
+    Notes
+    -----
+    ..[1] Zhao et. al, Convolutional neural networks for
+    time series classification, Journal of
+    Systems Engineering and Electronics, 28(1):2017.
+
+    Adapted from the implementation from Fawaz et. al
+    https://github.com/hfawaz/dl-4-tsc/blob/master/classifiers/cnn.py
+    """
+
+    def __init__(
+        self,
+        n_epochs=2000,
+        batch_size=16,
+        kernel_size=7,
+        avg_pool_size=3,
+        n_conv_layers=2,
+        callbacks=None,
+        verbose=False,
+    ):
```

Comment:
there should be a `_check_soft_dependencies` here too, at error severity level

Reply:
I've raised the severity to error, but was then wondering whether I could I not just do it in the base classes rather than in every individual file, I can try it out with the second one